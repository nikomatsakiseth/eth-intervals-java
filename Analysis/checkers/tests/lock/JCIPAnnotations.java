import net.jcip.annotations.*;

// Smoke test for supporting JCIP annotations
public class JCIPAnnotations {

    Object unguardedField;
    void unguardedMethod() { }

    Object lock;
    @GuardedBy("lock") Object guardedField;

    @GuardedBy("lock")
    void guardedMethod() { }

    void testUnguardedAccess() {
        this.unguardedField.toString();
        this.guardedField.toString();   // error
        this.unguardedMethod();
        this.guardedMethod();   // error
    }

    void testGuardedAccess() {
        synchronized(lock) {
            this.unguardedField.toString();
            this.guardedField.toString();
            this.unguardedMethod();
            this.guardedMethod();
        }
    }
}
