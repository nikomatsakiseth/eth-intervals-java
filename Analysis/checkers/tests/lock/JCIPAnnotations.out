:17: (unguarded.access)
:19: (unguarded.invocation)
